src/hotline/fname_binary_map.c

```diff
@@ -191,7 +218,15 @@ char *get_function_name(binary_info_t *info, uint64_t addr) {
           addr < info->symtab[i].st_value + info->symtab[i].st_size) {
         char *name = strdup(info->strtab + info->symtab[i].st_name);
         ASSERT(name != NULL, "Function name should not be null.");
-        return name;
+
+        char *demangled = demangle(name);
+        ASSERT(demangled != NULL, "Demangled function name should not be null.");
+
+        char *quoted_function = malloc(strlen(demangled) + 3);
+        sprintf(quoted_function, "\"%s\"", demangled);
```

Comment:
```suggestion
        size_t len = strlen(demangled);
        size_t capped_len = len > BUFSIZ ? BUFSIZ : len;
        char *quoted_function = malloc(capped_len + 3);
        snprintf(quoted_function, capped_len + 3, "\"%.*s\"", (int)capped_len, demangled);
```
`BUFSIZ` is defined in stdio.h as 8192 which seems like a more than generous limit which will prevent this from ever trying to run away with all system memory.